inc/saiipsec.h

```diff
@@ -59,14 +59,14 @@ typedef enum _sai_ipsec_cipher_t
  */
 typedef enum _sai_ipsec_sa_status_t
 {
-    /** SA sequence number above lower of 2 watermarks */
-    SAI_IPSEC_SA_STATUS_SN_ABOVE_LOW_WATERMARK,
+    /** SA byte count above lower of 2 watermarks */
+    SAI_IPSEC_SA_STATUS_ABOVE_LOW_WATERMARK,
 
-    /** SA sequence number above higher of 2 watermarks */
-    SAI_IPSEC_SA_STATUS_SN_ABOVE_HIGH_WATERMARK,
+    /** SA byte count above higher of 2 watermarks */
+    SAI_IPSEC_SA_STATUS_ABOVE_HIGH_WATERMARK,
 
-    /** SA sequence number at maximum limit */
-    SAI_IPSEC_SA_STATUS_SN_AT_MAX_LIMIT,
+    /** SA byte count reached supported maximum value */
```

Comment:
I think this one could be left at the SN_AT_MAX_LIMIT since it is covering the SN roll-over scenario. The LOW and HIGH watermarks map nicely to the hard and soft limits, so they are OK.

Reply:
So you want a 3-level (2 threshold) status for byte-count and only a max-limit status for SN?  But does not that leave you exposed to SN max-limit being reached while the byte-count thresholds have not been crossed?  If you really want to avoid SN reaching max limit, would it be better if there are independent threshold-based notification for SN also?

Reply:
The IPsec RFC specifies the hard and soft lifetime limits for byte based counts. Once the soft limit is reached, traffic continues but the notification is used by software to setup a new SA. Once the hard limit is reached, traffic must stop for the SA. Separately the RFC specifies that the SN must not rollover. Ideally I'd like these terms reflected in the status enum or at least have the comments relate it back to the RFC terms.

I'm OK with having an additional threshold for early SN notification but that is a nice to have since its not specified in the RFC.

Reply:
Since it is not necessary, I would keep it simple and not add extra thresholds for SN.  But I have another question.  
If we have a single status enum, I suppose when SN max-limit is reached that should be reported, even if some byte-count threshold is exceeded?  So we do not have need to separate SN status from byte-count status?

Reply:
I'm not 100% clear on what you are asking. Are you asking if both SN max limit and a lifetime limit are both hit on the same packet? If so, then ideally both notifications should be sent, or at a minimum, the most severe  - i.e. hard lifetime limit or SN max limit since both cause the SA to stop processing any more packets. In other words mixing SN status and byte-count status in the same enum is OK.

Reply:
I am going to update the API to use separate byte-count status and SN status in the struct sai_ipsec_sa_status_notification_t.  Please check if that meets your needs.

Reply:
The changes look good. Thanks!